### PR TITLE
Make seccomp annotations non-functional

### LINKED ISSUE #91286: Graduate Seccomp to GA
Graduate seccomp to GA, as defined on the respective [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190717-seccomp-ga.md). This is an umbrella issue to track all outstanding work.

### Core Items
- [x] API Server changes https://github.com/kubernetes/kubernetes/pull/91408 (MERGED)
- [x] Kubelet changes https://github.com/kubernetes/kubernetes/pull/91381 (MERGED)
- [x] Mark flag `--seccomp-profile-root` as deprecated https://github.com/kubernetes/kubernetes/pull/91182 (MERGED)
- [x] Update PSP to support seccomp fields https://github.com/kubernetes/kubernetes/pull/92856 (MERGED)
- [x] Document Kubernetes seccomp support https://github.com/kubernetes/website/pull/21278 (MERGED)

### Nice to Have on 1.19
- [x] Add seccomp least privilege for docker sandbox https://github.com/kubernetes/kubernetes/pull/90948 (MERGED)
- [x] Add seccomp least privilege for kuberuntime https://github.com/kubernetes/kubernetes/pull/90949 (MERGED)
- [x] E2E tests review https://github.com/kubernetes/kubernetes/pull/91442 (LGTM)

### Pushed to 1.22
- [x] Add Warning mechanism for deprecated features usage (based off [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1693-warnings))

/area kubelet
/area kubelet-api
/sig node
/area security
/assign

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind deprecation

#### What this PR does / why we need it:

This cleanup has been planned to finish the corresponding KEP: https://github.com/kubernetes/kubernetes/issues/91286

As follow-up on the partly removal of the seccomp annotations in https://github.com/kubernetes/kubernetes/pull/109819, we now drop the version skew handling completely, but still warn as well as keep the validation in place if both (annotation and field) are set.

The Pod Security Admission code has been already changed in https://github.com/kubernetes/kubernetes/pull/114846.

#### Which issue(s) this PR fixes:

Fixes #91286, https://github.com/kubernetes/enhancements/issues/135 

#### Special notes for your reviewer:

/cc @liggitt @tallclair 

#### Does this PR introduce a user-facing change?

```release-note
Action required: support for the alpha seccomp annotations `seccomp.security.alpha.kubernetes.io/pod` and `container.seccomp.security.alpha.kubernetes.io`, deprecated since v1.19, has been completely removed. The seccomp fields are no longer auto-populated when pods with seccomp annotations are created. Pods should use the corresponding pod or container `securityContext.seccompProfile` field instead.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
KEP: https://github.com/kubernetes/enhancements/issues/135
```

### CODE REVIEW

File: pkg/api/pod/warnings.go [hunk 1]
```diff
@@ -210,15 +210,15 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 	// use of pod seccomp annotation without accompanying field
 	if podSpec.SecurityContext == nil || podSpec.SecurityContext.SeccompProfile == nil {
 		if _, exists := meta.Annotations[api.SeccompPodAnnotationKey]; exists {
-			warnings = append(warnings, fmt.Sprintf(`%s: deprecated since v1.19, non-functional in a future release; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompPodAnnotationKey)))
+			warnings = append(warnings, fmt.Sprintf(`%s: deprecated since v1.19, non-functional since v1.27; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompPodAnnotationKey)))
```
Comment (liggitt) [inline pkg/api/pod/warnings.go:213]:
```suggestion
			warnings = append(warnings, fmt.Sprintf(`%s: non-functional in v1.27+; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompPodAnnotationKey)))
```

File: pkg/api/pod/warnings.go [hunk 2]
```diff
@@ -210,15 +210,15 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
...
 		}
 	}
 
 	pods.VisitContainersWithPath(podSpec, fieldPath.Child("spec"), func(c *api.Container, p *field.Path) bool {
 		// use of container seccomp annotation without accompanying field
 		if c.SecurityContext == nil || c.SecurityContext.SeccompProfile == nil {
 			if _, exists := meta.Annotations[api.SeccompContainerAnnotationKeyPrefix+c.Name]; exists {
-				warnings = append(warnings, fmt.Sprintf(`%s: deprecated since v1.19, non-functional in a future release; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompContainerAnnotationKeyPrefix+c.Name)))
+				warnings = append(warnings, fmt.Sprintf(`%s: deprecated since v1.19, non-functional since v1.27; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompContainerAnnotationKeyPrefix+c.Name)))
```
Comment (liggitt) [inline pkg/api/pod/warnings.go:221]:
```suggestion
				warnings = append(warnings, fmt.Sprintf(`%s: non-functional in v1.27+; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompContainerAnnotationKeyPrefix+c.Name)))
```

**Comment by liggitt** [comment]
nits on the message, lgtm otherwise

File: pkg/api/pod/warnings.go (hunk 2 above)
Comment (saschagrunert) [reply pkg/api/pod/warnings.go:221]: Fixed as suggested.

**Pull request merged**